### PR TITLE
Enable/Disable Query and Requests Caching with parameters

### PR DESCRIPTION
**Pull request by kiranprakash154**
### Description

**Providing ability to enable query and/or requests cache on a workload.**

pass the parameters below with true or false according to your need. 

```
query_cache_enabled
requests_cache_enabled
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by IanHoang — COMMENTED** [review]
Thanks for adding this parameter to give workloads better flexibility! Overall, the changes look good to me. Although it is unlikely, we want to ensure that changes to workloads do not cause breaking changes. Could you verify that there a no breaking changes by running a quick NYC Taxis test in `--test-mode` and pasting the output in the description above?

**Review by gkamat — CHANGES_REQUESTED** [review]
Please update the README.md to indicate this parameter is available to the user.  And also, run the workload with `--test-mode` as @IanHoang suggested.

**Comment by kiranprakash154** [comment]
@gkamat let me address your comment in a couple of days.

**Comment by kiranprakash154** [comment]
```
➜  nyc_taxis git:(kp/enable-cache-via-parameter) ✗ opensearch-benchmark execute-test --workload-path=$WORKLOAD_PATH --pipeline=benchmark-only --target-hosts=$DOMAIN_ENDPOINT --client-options="basic_auth_user:'admin',basic_auth_password:'Admin_123'" --test-mode

   ____                  _____                      __       ____                  __                         __
  / __ \____  ___  ____ / ___/___  ____ ___________/ /_     / __ )___  ____  _____/ /_  ____ ___  ____ ______/ /__
 / / / / __ \/ _ \/ __ \\__ \/ _ \/ __ `/ ___/ ___/ __ \   / __  / _ \/ __ \/ ___/ __ \/ __ `__ \/ __ `/ ___/ //_/
/ /_/ / /_/ /  __/ / / /__/ /  __/ /_/ / /  / /__/ / / /  / /_/ /  __/ / / / /__/ / / / / / / / / /_/ / /  / ,<
\____/ .___/\___/_/ /_/____/\___/\__,_/_/   \___/_/ /_/  /_____/\___/_/ /_/\___/_/ /_/_/ /_/ /_/\__,_/_/  /_/|_|
    /_/

[INFO] You did not provide an explicit timeout in the client options. Assuming default of 10 seconds.
[INFO] Executing test with workload [nyc_taxis], test_procedure [append-no-conflicts] and provision_config_instance ['external'] with version [1.3.2].

[WARNING] indexing_total_time is 1945 ms indicating that the cluster is not in a defined clean state. Recorded index time metrics may be misleading.
[WARNING] refresh_total_time is 423 ms indicating that the cluster is not in a defined clean state. Recorded index time metrics may be misleading.
Running delete-index                                                           [100% done]
Running create-index                                                           [100% done]
Running check-cluster-health                                                   [100% done]
Running index                                                                  [100% done]
Running refresh-after-index                                                    [100% done]
Running force-merge                                                            [100% done]
Running refresh-after-force-merge                                              [100% done]
Running wait-until-merges-finish                                               [100% done]
Running default                                                                [100% done]
Running range                                                                  [100% done]
Running distance_amount_agg                                                    [100% done]
Running autohisto_agg                                                          [100% done]
Running date_histogram_agg                                                     [100% done]

------------------------------------------------------
    _______             __   _____
   / ____(_)___  ____ _/ /  / ___/_________  ________
  / /_  / / __ \/ __ `/ /   \__ \/ ___/ __ \/ ___/ _ \
 / __/ / / / / / /_/ / /   ___/ / /__/ /_/ / /  /  __/
/_/   /_/_/ /_/\__,_/_/   /____/\___/\____/_/   \___/
------------------------------------------------------

|                                                         Metric |                     Task |       Value |   Unit |
|---------------------------------------------------------------:|-------------------------:|------------:|-------:|
|                     Cumulative indexing time of primary shards |                          |   0.0367167 |    min |
|             Min cumulative indexing time across primary shards |                          |           0 |    min |
|          Median cumulative indexing time across primary shards |                          |    0.005725 |    min |
|             Max cumulative indexing time across primary shards |                          |      0.0071 |    min |
|            Cumulative indexing throttle time of primary shards |                          |           0 |    min |
|    Min cumulative indexing throttle time across primary shards |                          |           0 |    min |
| Median cumulative indexing throttle time across primary shards |                          |           0 |    min |
|    Max cumulative indexing throttle time across primary shards |                          |           0 |    min |
|                        Cumulative merge time of primary shards |                          |           0 |    min |
|                       Cumulative merge count of primary shards |                          |           0 |        |
|                Min cumulative merge time across primary shards |                          |           0 |    min |
|             Median cumulative merge time across primary shards |                          |           0 |    min |
|                Max cumulative merge time across primary shards |                          |           0 |    min |
|               Cumulative merge throttle time of primary shards |                          |           0 |    min |
|       Min cumulative merge throttle time across primary shards |                          |           0 |    min |
|    Median cumulative merge throttle time across primary shards |                          |           0 |    min |
|       Max cumulative merge throttle time across primary shards |                          |           0 |    min |
|                      Cumulative refresh time of primary shards |                          |  0.00856667 |    min |
|                     Cumulative refresh count of primary shards |                          |          71 |        |
|              Min cumulative refresh time across primary shards |                          |           0 |    min |
|           Median cumulative refresh time across primary shards |                          |  0.00116667 |    min |
|              Max cumulative refresh time across primary shards |                          |  0.00228333 |    min |
|                        Cumulative flush time of primary shards |                          |           0 |    min |
|                       Cumulative flush count of primary shards |                          |           7 |        |
|                Min cumulative flush time across primary shards |                          |           0 |    min |
|             Median cumulative flush time across primary shards |                          |           0 |    min |
|                Max cumulative flush time across primary shards |                          |           0 |    min |
|                                        Total Young Gen GC time |                          |           0 |      s |
|                                       Total Young Gen GC count |                          |           0 |        |
|                                          Total Old Gen GC time |                          |           0 |      s |
|                                         Total Old Gen GC count |                          |           0 |        |
|                                                     Store size |                          |  0.00601556 |     GB |
|                                                  Translog size |                          | 4.09782e-07 |     GB |
|                                         Heap used for segments |                          |    0.062748 |     MB |
|                                       Heap used for doc values |                          |   0.0239677 |     MB |
|                                            Heap used for terms |                          |    0.026886 |     MB |
|                                            Heap used for norms |                          |  0.00366211 |     MB |
|                                           Heap used for points |                          |           0 |     MB |
|                                    Heap used for stored fields |                          |  0.00823212 |     MB |
|                                                  Segment count |                          |          17 |        |
|                                                 Min Throughput |                    index |     1336.25 | docs/s |
|                                                Mean Throughput |                    index |     1336.25 | docs/s |
|                                              Median Throughput |                    index |     1336.25 | docs/s |
|                                                 Max Throughput |                    index |     1336.25 | docs/s |
|                                        50th percentile latency |                    index |     792.145 |     ms |
|                                       100th percentile latency |                    index |     828.659 |     ms |
|                                   50th percentile service time |                    index |     792.145 |     ms |
|                                  100th percentile service time |                    index |     828.659 |     ms |
|                                                     error rate |                    index |           0 |      % |
|                                                 Min Throughput | wait-until-merges-finish |         5.6 |  ops/s |
|                                                Mean Throughput | wait-until-merges-finish |         5.6 |  ops/s |
|                                              Median Throughput | wait-until-merges-finish |         5.6 |  ops/s |
|                                                 Max Throughput | wait-until-merges-finish |         5.6 |  ops/s |
|                                       100th percentile latency | wait-until-merges-finish |     126.139 |     ms |
|                                  100th percentile service time | wait-until-merges-finish |     126.139 |     ms |
|                                                     error rate | wait-until-merges-finish |           0 |      % |
|                                                 Min Throughput |                  default |        7.08 |  ops/s |
|                                                Mean Throughput |                  default |        7.08 |  ops/s |
|                                              Median Throughput |                  default |        7.08 |  ops/s |
|                                                 Max Throughput |                  default |        7.08 |  ops/s |
|                                       100th percentile latency |                  default |     470.087 |     ms |
|                                  100th percentile service time |                  default |     328.747 |     ms |
|                                                     error rate |                  default |           0 |      % |
|                                                 Min Throughput |                    range |        6.05 |  ops/s |
|                                                Mean Throughput |                    range |        6.05 |  ops/s |
|                                              Median Throughput |                    range |        6.05 |  ops/s |
|                                                 Max Throughput |                    range |        6.05 |  ops/s |
|                                       100th percentile latency |                    range |     365.162 |     ms |
|                                  100th percentile service time |                    range |     199.638 |     ms |
|                                                     error rate |                    range |           0 |      % |
|                                                 Min Throughput |      distance_amount_agg |        4.85 |  ops/s |
|                                                Mean Throughput |      distance_amount_agg |        4.85 |  ops/s |
|                                              Median Throughput |      distance_amount_agg |        4.85 |  ops/s |
|                                                 Max Throughput |      distance_amount_agg |        4.85 |  ops/s |
|                                       100th percentile latency |      distance_amount_agg |     506.237 |     ms |
|                                  100th percentile service time |      distance_amount_agg |     299.621 |     ms |
|                                                     error rate |      distance_amount_agg |           0 |      % |
|                                                 Min Throughput |            autohisto_agg |        5.11 |  ops/s |
|                                                Mean Throughput |            autohisto_agg |        5.11 |  ops/s |
|                                              Median Throughput |            autohisto_agg |        5.11 |  ops/s |
|                                                 Max Throughput |            autohisto_agg |        5.11 |  ops/s |
|                                       100th percentile latency |            autohisto_agg |     503.212 |     ms |
|                                  100th percentile service time |            autohisto_agg |     307.224 |     ms |
|                                                     error rate |            autohisto_agg |           0 |      % |
|                                                 Min Throughput |       date_histogram_agg |        4.81 |  ops/s |
|                                                Mean Throughput |       date_histogram_agg |        4.81 |  ops/s |
|                                              Median Throughput |       date_histogram_agg |        4.81 |  ops/s |
|                                                 Max Throughput |       date_histogram_agg |        4.81 |  ops/s |
|                                       100th percentile latency |       date_histogram_agg |     304.909 |     ms |
|                                  100th percentile service time |       date_histogram_agg |     96.3521 |     ms |
|                                                     error rate |       date_histogram_agg |           0 |      % |


--------------------------------
[INFO] SUCCESS (took 17 seconds)
--------------------------------
```

**Review by IanHoang — APPROVED** [review]

**Review by IanHoang — CHANGES_REQUESTED** [review]
Could you quickly add this parameter and its associated definition and default value to each README?

File: percolator/README.md [hunk 1]
```diff
@@ -25,6 +25,9 @@ This workload allows the following parameters to be specified using `--workload-
 * `ingest_percentage` (default: 100): A number between 0 and 100 that defines how much of the document corpus should be ingested.
 * `number_of_replicas` (default: 0)
 * `number_of_shards` (default: 5)
+* `query_cache_enabled` (default: false)
+* `requests_cache_enabled` (default: false)
+*
```
Comment (IanHoang) [inline percolator/README.md:30]: Nit: Extra space

**Review by IanHoang — APPROVED** [review]
Thanks for the contribution!

File: percolator/README.md (hunk 1 above)
Comment (kiranprakash154) [reply percolator/README.md]: Thanks for catching it, addressed.

**Comment by kiranprakash154** [comment]
@gkamat Looks like i'm blocked from merging this until I have your 👍 
if you're okay with everything, feel free to approve and merge it. 
Thanks !

**Review by gkamat — APPROVED** [review]

**Pull request merged**